### ISSUE TITLE
I have a error, method Schedule is not mocked

### ISSUE THREAD

**Issue by christ998** [dariosn85/ts-jest-mocker#6]
This is my code
```
let portalService: VirtualPortalService;
  let httpService: HttpService;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        VirtualPortalService,
        {
          provide: HttpService,
          useValue: mock<HttpService>(HttpService),
        },
      ],
    }).compile();

    portalService = module.get<VirtualPortalService>(VirtualPortalService);
    httpService = module.get<HttpService>(HttpService);
  });

it('should send a POST request to the logout endpoint with the provided rut and token', async () => {
      const rut = '123456789';
      const token = 'abc123';
      const axiosResponseMock = mock<AxiosResponse<ILogoutResponse>>();
      const expectedResponse = {
        payload: '{"logout":true}',
      };

      jest.spyOn(httpService, 'post').mockImplementationOnce(() => of(axiosResponseMock));

      const result = await portalService.logout(rut, token);

      expect(result).toEqual(expectedResponse);
    });
```
When I run the test, it shows Error: Method schedule is not mocked

**Comment by christ998** [comment]
@dariosn85 could you help me with this?

**Comment by dariosn85** [comment]
@christ998 Probably there is `schedule` method in the `ILogoutResponse` or in `HttpService`, and you didn't mock the default behavior to return a value or override implementation when method is called.

The default implementation (if you forget to mock something that is called) will throw [this error](https://github.com/dariosn85/ts-jest-mocker/blob/main/src/lib/proxy-generic.ts#L11).

**Comment by christ998** [comment]
@dariosn85 Thanks for your response. I don't have any schedule method in ILogoutResponse. I'm working with Nestjs. This is my code
```
interface VirtualPortalResponse<T> {
  payload: T;
  token?: string;
}
interface ILogoutPayload {
  logout: boolean;
}
export interface ILogoutResponse extends VirtualPortalResponse<ILogoutPayload> {}
```
HttpService is the service of HttpModule, I don't know if HttpService has a schedule method, I don't think so

**Comment by dariosn85** [comment]
@christ998 do you see stack trace in the output (to understand from where `schedule` is called)?

**Comment by christ998** [comment]
@dariosn85 Here I show you the log in catchError method in the pipe method
```
Error: Method schedule is not mocked
        at Proxy.<anonymous> (node_modules\ts-jest-mocker\lib\lib\proxy-generic.js:11:23)
        at node_modules\jest-mock\build\index.js:397:39
        at Proxy.<anonymous> (node_modules\jest-mock\build\index.js:404:13)
        at Proxy.mockConstructor (node_modules\jest-mock\build\index.js:148:19)
        at Observable._subscribe (node_modules\rxjs\src\internal\scheduled\scheduleArray.ts:9:22)
        at Observable.Object.<anonymous>.Observable._trySubscribe (node_modules\rxjs\src\internal\Observable.ts:244:19)
        at node_modules\rxjs\src\internal\Observable.ts:234:18
        at Object.errorContext (node_modules\rxjs\src\internal\util\errorContext.ts:29:5)
        at Observable.Object.<anonymous>.Observable.subscribe (node_modules\rxjs\src\internal\Observable.ts:220:5)
        at node_modules\rxjs\src\internal\operators\catchError.ts:115:23
        at OperatorSubscriber.<anonymous> (node_modules\rxjs\src\internal\util\lift.ts:24:18)
        at node_modules\rxjs\src\internal\Observable.ts:226:22
        at Object.errorContext (node_modules\rxjs\src\internal\util\errorContext.ts:29:5)
        at Observable.Object.<anonymous>.Observable.subscribe (node_modules\rxjs\src\internal\Observable.ts:220:5)
        at node_modules\rxjs\src\internal\operators\map.ts:54:12
        at SafeSubscriber.<anonymous> (node_modules\rxjs\src\internal\util\lift.ts:24:18)
        at node_modules\rxjs\src\internal\Observable.ts:226:22
        at Object.errorContext (node_modules\rxjs\src\internal\util\errorContext.ts:29:5)
        at Observable.Object.<anonymous>.Observable.subscribe (node_modules\rxjs\src\internal\Observable.ts:220:5)
        at node_modules\rxjs\src\internal\firstValueFrom.ts:73:12
        at new Promise (<anonymous>)
        at firstValueFrom (node_modules\rxjs\src\internal\firstValueFrom.ts:58:10)
        at ApiSbpayService.logout (src\apis\api-sbpay\api-sbpay.service.ts:14:45)
        at Object.<anonymous> (src\apis\api-sbpay\api-sbpay.service.spec.ts:64:42)
        at Promise.then.completed (node_modules\jest-circus\build\utils.js:298:28)
        at new Promise (<anonymous>)
        at callAsyncCircusFn (node_modules\jest-circus\build\utils.js:231:10)
        at _callCircusTest (node_modules\jest-circus\build\run.js:316:40)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at _runTest (node_modules\jest-circus\build\run.js:252:3)
        at _runTestsForDescribeBlock (node_modules\jest-circus\build\run.js:126:9)
        at _runTestsForDescribeBlock (node_modules\jest-circus\build\run.js:121:9)
        at _runTestsForDescribeBlock (node_modules\jest-circus\build\run.js:121:9)
        at run (node_modules\jest-circus\build\run.js:71:3)
        at runAndTransformResultsToJestFormat (node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapterInit.js:122:21)
        at jestAdapter (node_modules\jest-circus\build\legacy-code-todo-rewrite\jestAdapter.js:79:19)
        at runTestInternal node_modules\jest-runner\build\runTest.js:367:16)
        at runTest (node_modules\jest-runner\build\runTest.js:444:34)
```

**Comment by dariosn85** [comment]
@christ998 

It took me a while to understand what is going on. The "issue" is related to the `rxjs` in your case. This happened:

1. Mock was created (this line `const axiosResponseMock = mock<AxiosResponse<ILogoutResponse>>();`). Here generic `Proxy` class is created. This proxy will return "method not mocked" error in case if method that is not mocked is used.
2. In the line `jest.spyOn(httpService, 'post').mockImplementationOnce(() => of(axiosResponseMock));` you created `Observable` using `of()` utility function from `rxjs`. This function internally checks if so called `Scheduler` was passed as a parameter. This is done by checking if passed parameter has `.schedule` function. As Proxy class mocks everything it will also return something for `.scheduler` method, so this condition is fulfilled. 
3. So in such case Proxy (mock) + `rxjs` caused that issue

The minimal example how to reproduce this (without `ts-jest-mocker`):

```typescript
const object = {
    schedule: () => {
        // if you throw error, code below will fail
        // if you don't throw error, code will hang as not real rxjs Scheduler code will be called
        throw new Error('Test error');
    }
};

const observable = of(object);

const value = await firstValueFrom(observable);

console.log(value);
```

I will think about it from the `ts-jest-mocker` perspective and implement more friendly code.

Until then you could go with a workaround: provide axios response directly, not mocked object.

```typescript
let httpService: Mock<HttpService>;

// ... the rest of the code, beforeEach,...

const axiosResponseMock = {
    data: {
        loggedOut: true,
    }, // data/object that your logout API returns
} as AxiosResponse; // we are not using mock() here

httpService.post.mockReturnValueOnce(of(axiosResponseMock));
```

Update: while investigating error I found there is also reported bug in `rxjs` for that https://github.com/ReactiveX/rxjs/issues/6584. But using `from` instead `of` will currently not work with `ts-jest-mocker`. The reason for that is similar thing, but for `then` method (Promise). I have a few ideas what to do and will try to provide update in the next days.

Let me know if the suggested workaround will work for you.

**Comment by dariosn85** [comment]
@christ998 There is new version available v1.1.0 with which you can also handle the above issue. See this section: https://github.com/dariosn85/ts-jest-mocker/?tab=readme-ov-file#ts-jest-mocker-with-rxjs

For now I'm closing the issue, but let me know if there will be anything else.
